consensus/hotstuff/validator/validator.go

```diff
@@ -129,63 +198,75 @@ func (v *Validator) ValidateProposal(proposal *model.Proposal) error {
 		return newInvalidBlockError(block, fmt.Errorf("proposer %s is not leader (%s) for view %d", block.ProposerID, leader, block.View))
 	}
 
-	// check the proposer is a valid committee member at the most recent state on this fork
-	// TODO: move this check into SafetyRules when front-loading the verification
-	_, err = v.committee.IdentityByBlock(block.BlockID, block.ProposerID)
-	if model.IsInvalidSignerError(err) {
-		return newInvalidBlockError(block, fmt.Errorf("proposer %x is leader but is not a valid committee member", block.ProposerID))
+	// The Block must contain a proof that the primary legitimately entered the respective view.
+	// Transitioning to proposal.Block.View is possible either by observing a QC or a TC for the
+	// previous round. If and only if the QC is _not_ for the previous round we require a TC for
+	// the previous view to be present.
+	lastViewSuccessful := proposal.Block.View == proposal.Block.QC.View+1
+	if !lastViewSuccessful {
+		// check if proposal is correctly structured
+		if proposal.LastViewTC == nil {
+			return newInvalidBlockError(block, fmt.Errorf("QC in block is not for previous view, so expecting a TC but none is included in block"))
+		}
+
+		// check if included TC is for previous view
+		if proposal.Block.View != proposal.LastViewTC.View+1 {
+			return newInvalidBlockError(block, fmt.Errorf("QC in block is not for previous view, so expecting a TC for view %d but got TC for view %d", proposal.Block.View-1, proposal.LastViewTC.View))
+		}
+
+		// Check if proposal extends either the newest QC specified in the TC, or a newer QC
+		// in edge cases a leader may construct a TC and QC concurrently such that TC contains
+		// an older QC - in these case we still want to build on the newest QC, so this case is allowed.
+		if proposal.Block.QC.View < proposal.LastViewTC.NewestQC.View {
+			return newInvalidBlockError(block, fmt.Errorf("TC in block contains a newer QC than the block itself, which is a protocol violation"))
+		}
+	} else if proposal.LastViewTC != nil {
+		// last view ended with QC, including TC is a protocol violation
+		return newInvalidBlockError(block, fmt.Errorf("last view has ended with QC but proposal includes LastViewTC"))
 	}
+
+	// Check signatures, keep the most expensive the last to check
+
+	// check if included QC is valid
+	err = v.ValidateQC(qc)
 	if err != nil {
-		return fmt.Errorf("unexpected error checking proposer standing for block %x: %w", block.BlockID, err)
+		if model.IsInvalidQCError(err) {
+			return newInvalidBlockError(block, fmt.Errorf("invalid qc included: %w", err))
+		}
```

Comment:
same argument here:
```suggestion
		}
				if errors.Is(err, model.ErrViewForUnknownEpoch) {
			// We require each replica to be bootstrapped with a QC pointing to a finalized block. Consensus safety rules guarantee that
			// a QC at least as new as the root QC must be contained in any TC. This is because the TC must include signatures from a
			// supermajority of replicas, including at least one honest replica, which attest to their locally highest known QC. Hence, 
			// any QC included in a TC must be the root QC or newer. Therefore, we should know the Epoch for any QC we encounter.
			// receiving a `model.ErrViewForUnknownEpoch` is conceptually impossible, i.e. a symptom of an internal bug or invalid
			// bootstrapping information. 
			return fmt.Errorf("no Epoch information availalbe for QC that was included in TC; symptom of internal bug or invalid bootstrapping information: %s", err.Error())
		}
```